File: WebApplication1/wwwroot/js/Maquila.js
$(document).ready(function () {
    dataTable = $("#tblMaquila").DataTable({
        "ajax": {
            "url": "/Maquilas/GetTodosMaquila",
            "type": "GET",
            "dataType": "json",
        },
        "columns": [
            { "data": "idMaquila", "width": "10%" },
            { "data": "modelo", "width": "20%" },
            { "data": "maquilero1", "width": "10%" },
            { "data": "maquilero2", "width": "10%" },
            { "data": "maquilero3", "width": "10%" },
            { "data": "maquilero4", "width": "10%" },
            { "data": "fechaEntregaMaq1", "width": "30%", "render": function (data) { return moment(data).format('DD/MM/YYYY') } },
            {
                "data": "fechaEntregaMaq2", "width": "30%", "render": function (data) {
                    return data ? moment(data).format('DD/MM/YYYY') : "Sin fecha";
                }
            },
            {
                "data": "fechaEntregaMaq3", "width": "30%", "render": function (data) {
                    return data ? moment(data).format('DD/MM/YYYY') : "Sin fecha";
                }
            },
            {
                "data": "fechaEntregaMaq4", "width": "30%", "render": function (data) {
                    return data ? moment(data).format('DD/MM/YYYY') : "Sin fecha";
                }
            },
            { "data": "fechaMaquila", "width": "30%", "render": function (data) { return moment(data).format('DD/MM/YYYY') } },
            { "data": "fechaRecepcion", "width": "30%", "render": function (data) { return moment(data).format('DD/MM/YYYY') } },
            {
                "data": "fechaEntrega", "width": "30%", "render": function (data) {
                    return data ? moment(data).format('DD/MM/YYYY') : "Sin fecha";
                }
            },
            { "data": "cantidadRecibida", "width": "30%" },
            { "data": "incidencias", "width": "30%"},
            { "data": "area", "width": "30%" },
            {
                "data": "idMaquila",
                "render": function (data, type, row) {
                    // Comprobar si el registro tiene fecha de entrega
                    let btnClass = row.fechaEntrega ? "btn-secondary" : "btn-success";
                    let btnText = row.fechaEntrega ? "Editado" : "Editar";
                    return `<div class="text-center">
                        <a href="/Maquilas/Edit/${data}" class="btn ${btnClass} text-white" style="cursor: pointer;">
                            <i class="fas fa-edit"></i> ${btnText}
                        </a>
                    </div>`;
                },
                "width": "20%"
            }
        ],
        "language": {
            "search": "Buscar:",
        },
        "dom": '<"top"f>rt<"bottom"ip><"clear">',
        "searching": true,
        "scrollX": true,
        "paging": false,
        "info": false,
    });
    $('#tblMaquila_filter').detach().appendTo('.d-flex');
});

File: WebApplication1/Models/Maquila.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebTabitas.Models
{
    public class Maquila
    {
        public int IdMaquila { get; set; }
        public int IdGeneral { get; set; }
        public string Modelo { get; set; }
        [Required(ErrorMessage = "El Maquilero es obligatorio")]
        public string Maquilero1 { get; set; }
        public string? Maquilero2 { get; set; }
        public string? Maquilero3 { get; set; }
        public string? Maquilero4 { get; set; }
        [Required(ErrorMessage = "La fecha de Maquilero es obligatorio")]
        public DateTime FechaEntregaMaq1 { get; set; } = DateTime.Now;
        public DateTime? FechaEntregaMaq2 { get; set; } 
        public DateTime? FechaEntregaMaq3 { get; set; }
        public DateTime? FechaEntregaMaq4 { get; set; }
        [Required(ErrorMessage = "La Fecha de Maquila es obligatorio")]
        public DateTime FechaMaquila { get; set; } = DateTime.Now;
        [Required(ErrorMessage = "La Fecha de Recepcion es obligatorio")]
        public DateTime FechaRecepcion { get; set; } = DateTime.Now;
        public DateTime? FechaEntrega { get; set; }
        [Required(ErrorMessage = "La Cantidad es obligatorio")]
        public int? CantidadRecibida { get; set; }
        public string? Incidencias { get; set; }
        public int IdProceso { get; set; } 
        public string Area { get; set; }
    }
}
